src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -527,19 +529,19 @@ int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
 
 
 static int uct_rc_mlx5_iface_srq_topo(uct_rc_mlx5_iface_common_t *iface,
-                                      uct_md_h md,
-                                      uct_rc_mlx5_iface_common_config_t *mlx5_config)
+                                      uct_rc_mlx5_iface_common_config_t *mlx5_config,
+                                      int dc)
 {
-    uct_ib_mlx5_md_t *ib_md = ucs_derived_of(md, uct_ib_mlx5_md_t);
-
-    /* Cyclic SRQ is supported with DEVX only if: */
-    if (((mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_AUTO) ||
-        (mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_CYCLIC)) &&
-        (ib_md->flags & UCT_IB_MLX5_MD_FLAG_DEVX) &&
-        (/* - either HW TM is enabled */
-         UCT_RC_MLX5_TM_ENABLED(iface) ||
-         /* - or DevX RMP object is supported for SW TM */
-         (ib_md->flags & UCT_IB_MLX5_MD_FLAG_RMP))) {
+    int srq_cyclic_reqiured  = (mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_CYCLIC);
+    int srq_cyclic_supported = uct_rc_mlx5_is_devx_srq_supported(iface, 1);
+
+    if (srq_cyclic_reqiured ||
+        (mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_AUTO)) {
+        if (!srq_cyclic_supported && srq_cyclic_reqiured) {
+            ucs_error("Cyclic SRQ WQ type was requested, but it is not "
+                      "supported by the configuration");
+            return UCT_IB_MLX5_SRQ_TOPO_UNKNOWN;
```

Comment:
instead of adding UCT_IB_MLX5_SRQ_TOPO_UNKNOWN, return topo by pointer, and return ucs_status_t - and here it will be UNSUPPORTED

Reply:
done